connectd/connect.c

```diff
@@ -882,7 +889,7 @@ struct io_plan *connection_out(struct io_conn *conn, struct reaching *reach)
 
 	reach->connstate = "Cryptographic handshake";
 	return initiator_handshake(conn, &reach->daemon->id, &reach->id,
-				   &reach->addr,
+				   &reach->addrs[reach->addrnum],
```

Comment:
Are we sure this cannot overrun? `% tal_count(reach->addrs)` seems to be a good idea here.